File: utils/go/prime-sieve.go
package utils

func NewPrimeSieve(size int) []bool {

	// Create prime sieve
	prime_sieve := make([]bool, size)

	// Populate the prime sieve to assume all numbers are prime
	for i := range size {
		prime_sieve[i] = true
	}

	// 0 and 1 are not primes
	prime_sieve[0] = false
	prime_sieve[1] = false

	// Set all composite numbered indexes in the sieve to false
	i := 2
	for i < size {
		for index := i * i; index < size; index += i {
			prime_sieve[index] = false
		}

		// Update i to the next prime exposed by the sieve
		i++
		for i < size && !prime_sieve[i] {
			i++
		}
	}

	return prime_sieve
}

File: utils/prime_sieve.py
def newPrimeSieve(size):
    # Create prime sieve and populate to assume all numbers are prime, initially
    prime_sieve = [True] * size

    i = 2
    while i < size:
        for index in range(i*i, size, i):
            prime_sieve[index] = False

        # Update i to the next prime exposed by the sieve
        i += 1
        while i < size and not prime_sieve[i]:
            i += 1
    
    return prime_sieve